pages/publicBasket.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Card, Container, Header, Image,
+} from 'semantic-ui-react';
+import { getCurrentBasket } from '../utils/data/mergedData';
+import PublicVegCard from '../components/PublicVegCard';
+
+function PublicBasket() {
+  const [basket, setBasket] = useState({});
+
+  useEffect(() => {
+    getCurrentBasket().then(setBasket);
```

Comment:
I see that your function here ends up making an axios call to a JSON file, but regardless of where you're getting the data - loading any async call in a `useEffect` should be avoided (mostly) at all costs - Get rid of the `useEffect` is my opinion. 

Your CRUD actions are creating new `Promises` around `axios`, which does this for you already OOTB. Long story short is you're resolving 2 or 3 more promise layers than you need. Just rely on a single `axios` call to resolve or reject your API calls. If this was `production` you'd have to handle errors for every promise layer - not fun and extremely error prone.

Ex:
```js
const getActiveBasket = async () => {
   let data;
   try {
    data = axios.get(url).then((res) => res)
    }
   catch(e) { 
      return e.error;
    }

   if (!data) return false; // indicates an error state or just an empty cart 

  // otherwise return the cart
   return data
};
```

This feels redundant, but then when you use this function, you give yourself a possible error state to handle in the UI - making the integration more resilient 

ex: 
```js
const basket = getCurrentBasket();

if (!basket) return <SomeComponent /> // probably not an actual error for the UI - just do nothing

return // something that uses your `basket` variable - no need for a `useState` here

Reply:
I'm not sure I can do that due to using Firebase.  Unless I'm not aware, I can't query related data in the same call.   Want to discuss this.

Reply:
You can do anything!

If I have time I'll look, but essentially what you need is a query client - 💯 so if you're using an external DB - something like `react-query` used to exist (I think it changed not that long ago). I'd start there and see what you can find.

Reply:
FWIW - Firebase is the most overkill solution for this project. How are you hosting it?